### ISSUE TITLE
`println_empty_string`: span is off, too much code gets removed

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#16843]
Using the following flags

`--force-warn clippy::println_empty_string`

this code:

```rust
use std::io::Write;

fn main() -> Result<(), std::io::Error> {

    println!{""}

    Ok(())
}

```

caused the following diagnostics:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.YZKR4Pu3mFuB/icemaker_clippyfix_tempdir.UwdEp3BU9msQ/_a)
warning: empty string literal in `println!`
 --> src/main.rs:5:5
  |
5 |     println!{""}
  |     ^^^^^^^^^---
  |              |
  |              help: remove the empty string
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#println_empty_string
  = note: requested on the command line with `--force-warn clippy::println-empty-string`

warning: `_a` (bin "_a") generated 1 warning (run `cargo clippy --fix --bin "_a" -p _a -- -Aclippy::all -Awarnings --force-warn clippy::println_empty_string --cap-lints warn` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s

```

However after applying these diagnostics, the resulting code:

```rust
use std::io::Write;

fn main() -> Result<(), std::io::Error> {

    println!{

    Ok(())
}

```
no longer compiled:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.YZKR4Pu3mFuB/icemaker_clippyfix_tempdir.UwdEp3BU9msQ/_a)
error: this file contains an unclosed delimiter
 --> src/main.rs:8:3
  |
3 | fn main() -> Result<(), std::io::Error> {
  |                                         - unclosed delimiter
4 |
5 |     println!{
  |             - this delimiter might not be properly closed...
...
8 | }
  | -^
  | |
  | ...as it matches this but it has different indentation

error: could not compile `_a` (bin "_a" test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_a` (bin "_a") due to 1 previous error

```

Version:
```
rustc 1.97.0-nightly (bf4fbfb7a 2026-04-11)
binary: rustc
commit-hash: bf4fbfb7a18d74e7cd8eef93af7329c58fbb5344
commit-date: 2026-04-11
host: x86_64-unknown-linux-gnu
release: 1.97.0-nightly
LLVM version: 22.1.2

```
